src/Agent/NewRelic/Agent/Core/Attributes/AttributeValueCollection.cs

```diff
@@ -275,6 +288,18 @@ public void MakeImmutable()
 
             IsImmutable = true;
         }
+
+        private void LogTransactionIfFinest(string message)
```

Comment:
I would suggest just logging the transaction guid always. Also, when would there not be a transaction guid?

Reply:
There are several different constructors for this class and only one of them (the new one) takes in the guid which is used by TransactionMetadata.   The AttributeValueCollection class is used by many different parts of the agent that don't have access to the guid so this change is limited to just the transaction bit to keep it from spider-webbing.